doc/releases/changelog-dev.md

```diff
@@ -179,20 +179,24 @@
 
 <h4>Compile Pipeline and Transforms </h4>
 
-* The `TransformProgram` has been renamed to :class:`~.CompilePipeline`, and is now available at the
-  top level as `qml.CompilePipeline`. Uses of the term "transform program" has been updated to 
-  "compile pipeline" across the codebase. The class is still accessible as `TransformProgram` from 
-  `pennylane.transforms.core`, but the module `pennylane.transforms.core.transform_program`
-  has been renamed to `pennylane.transforms.core.compile_pipeline`, and the old name is no longer available.
+* A new :class:`~.CompilePipeline` class (previously known as the `TransformDispatcher`) is now available
+  at the top level as `qml.CompilePipeline`. Using this class you can now define large and complex
+  compilation passes in an intuitive and flexible way. For backward compatibility, the `TransformProgram`
+  class can still be accessed from `pennylane.transforms.core`.
   [(#8735)](https://github.com/PennyLaneAI/pennylane/pull/8735)
 
-* The `TransformDispatcher` has been renamed to :class:`~.transforms.core.Transform` and is now
+* For naming consistency, uses of the term "transform program" have been updated to "compile pipeline" across the codebase.
+  Correspondingly, the module `pennylane.transforms.core.transform_program` has been renamed to
+  `pennylane.transforms.core.compile_pipeline`, and the old name is no longer available.
+  [(#8735)](https://github.com/PennyLaneAI/pennylane/pull/8735)
+
+* The ``TransformDispatcher`` class has been renamed to :class:`~.transforms.core.Transform` and is now
   available at the top level as `qml.transform`.
   [(#8756)](https://github.com/PennyLaneAI/pennylane/pull/8756)
```

Comment:
I forgot to mention but the old name is still available as an alias for now.

Reply:
Include it here https://github.com/PennyLaneAI/pennylane/pull/8818/commits/0687af56fa79c1d0f3293ff67dbec4a6dcfe47ae

However I am wondering if this is the right way of phrasing these renaming changes since this was internal functionality. Would we have to make deprecation cicles for `TransformProgram` and `TransformDispatcher` latter on if we advertise them here in the changelog?

Reply:
I don't think it's worth doing a whole deprecation cycle for them because as you said they are after all internal.